playbooks/roles/openconnect/tasks/main.yml

```diff
@@ -23,71 +23,26 @@
   args:
     creates: /etc/rc0.d/K20ocserv
 
-- name: Generate the CA template file
-  template:
-    src: ca.tmpl.j2
-    dest: "{{ ocserv_ca_template_file }}"
-    owner: root
-    group: root
-    mode: 0600
-
-- name: Generate the CA key
-  command: certtool --generate-privkey --outfile {{ ocserv_ca_key_file }}
-  args:
-    creates: "{{ ocserv_ca_key_file }}"
-
-- name: Generate the self-signed CA certificate
-  command: certtool --generate-self-signed --load-privkey {{ ocserv_ca_key_file }} --template {{ ocserv_ca_template_file }} --outfile {{ ocserv_ca_certificate_file }}
-  args:
-    creates: "{{ ocserv_ca_certificate_file }}"
-
-- name: Generate the server template file
-  template:
-    src: server.tmpl.j2
-    dest: "{{ ocserv_server_template_file }}"
-    owner: root
-    group: root
-    mode: 0600
-
-- name: Generate the server key
-  command: certtool --generate-privkey --outfile {{ ocserv_server_key_file }}
-  args:
-    creates: "{{ ocserv_server_key_file }}"
-
-- name: Generate the server certificate
-  command: certtool --generate-certificate --load-privkey {{ ocserv_server_key_file }} --load-ca-certificate {{ ocserv_ca_certificate_file }} --load-ca-privkey {{ ocserv_ca_key_file }} --template {{ ocserv_server_template_file }} --outfile {{ ocserv_server_certificate_file }}
-  args:
-    creates: "{{ ocserv_server_certificate_file }}"
-
-- name: Generate the client template file
-  template:
-    src: client.tmpl.j2
-    dest: "{{ ocserv_path }}/client-{{ client_name.stdout }}.tmpl"
-    owner: root
-    group: root
-    mode: 0600
-  with_items: "{{ vpn_client_names.results }}"
-  loop_control:
-    loop_var: "client_name"
-    label: "{{ client_name.item }}"
-
-- name: Generate the client keys
-  command: certtool --generate-privkey --outfile {{ ocserv_path }}/{{ client_name.stdout }}-key.pem
-  args:
-    creates: "{{ ocserv_path }}/{{ client_name.stdout }}-key.pem"
+- name: Create directories for clients
+  file:
+    path: "{{ ocserv_path}}/{{ client_name.stdout }}"
+    state: directory
   with_items: "{{ vpn_client_names.results }}"
   loop_control:
     loop_var: "client_name"
     label: "{{ client_name.item }}"
 
-- name: Generate the client certificates
-  command: certtool --generate-certificate --load-privkey {{ ocserv_path }}/{{ client_name.stdout }}-key.pem --load-ca-certificate {{ ocserv_ca_certificate_file }} --load-ca-privkey {{ ocserv_ca_key_file }} --template {{ ocserv_path }}/client-{{ client_name.stdout }}.tmpl --outfile {{ ocserv_path }}/{{ client_name.stdout }}-cert.pem
-  args:
-    creates: "{{ ocserv_path }}/{{ client_name.stdout }}-cert.pem"
-  with_items: "{{ vpn_client_names.results }}"
-  loop_control:
-    loop_var: "client_name"
-    label: "{{ client_name.item }}"
+- include_role:
```

Comment:
I think this will need to be `import_playbook` for Ansible 2.4.0+

Reply:
Fails in [CI](https://travis-ci.org/alimakki/streisand/builds/288296374) if changed

Reply:
Does this `include_role` generate a warning in 2.4.0?

Reply:
Nope

Reply:
Then I'm satisfied :-)

Reply:
FWIW even the download-and-verify stuff weren't touched as part of the Ansible 2.4 update